### PR TITLE
Cleanup and testing

### PR DESCRIPTION
**Pull request by jonapap**
- Adds the first tests. 
  - Those tests are run on the board and will interact with the actual peripherals.
- Switch flashing to `probe-run`

### CODE REVIEW

File: libraries/common-arm/src/sd.rs [hunk 1]
```diff
@@ -1,14 +1,33 @@
 use core::marker::PhantomData;
```
Comment (jonapap) [inline libraries/common-arm/src/sd.rs:1]: Some linter fixes which I missed in Noah's PR

Comment (NoahSprenger) [reply libraries/common-arm/src/sd.rs:1]: I missed these, thank you for fixing.

File: libraries/messages/src/lib.rs [hunk 2]
```diff
@@ -17,6 +17,9 @@ use ts_rs::TS;
 
 pub mod sender;
 pub mod sensor;
+
+pub const MAX_SIZE: usize = 85;
```
Comment (NoahSprenger) [inline libraries/messages/src/lib.rs:21]: Why 85?

**Review by NoahSprenger — APPROVED** [review]
Overall looks good.

File: libraries/messages/src/lib.rs (hunk 2 above)
Comment (jonapap) [reply libraries/messages/src/lib.rs:24]: This is currently the max buffer size that is obtained when serializing the sbg struct.

**Pull request merged**